### ISSUE TITLE
Consent search by patient reference appears broken

### ISSUE THREAD

**Issue by julian-raymar** [FirelyTeam/spark#329]
**Description**
When searching for Consent objects the `patient` query param seems to be broken. Using the initial spark dataset, a search for Consents with the patient `Patient/96bca740-1642-455d-af02-48a83f785b53` yields 0 results.

**To Reproduce**
1. Bring up the spark service locally on port 5555
2. Perform a search for a Consent object http://localhost:5555/fhir/Consent?patient=Patient/96bca740-1642-455d-af02-48a83f785b53 (This Consent record is part of the initial spark dataset or seed data, so it should be present when the service starts up)
3. Observe the returned results
```json
{
  "resourceType": "Bundle",
  "id": "f4468d37-f402-411e-b444-3e75be2795e8",
  "type": "searchset",
  "total": 0,
  "link": [
    {
      "relation": "self",
      "url": "http://localhost:5555/fhir/_snapshot?id=c5c878e1-b6fc-4c02-84f0-01d2b83c7dc1&start=0"
    },
    {
      "relation": "first",
      "url": "http://localhost:5555/fhir/_snapshot?id=c5c878e1-b6fc-4c02-84f0-01d2b83c7dc1&start=0"
    },
    {
      "relation": "last",
      "url": "http://localhost:5555/fhir/_snapshot?id=c5c878e1-b6fc-4c02-84f0-01d2b83c7dc1&start=0"
    }
  ]
}
```

**Expected behavior**
Consent objects with only the supplied patient id should be present in the result set, rather than an empty result set. What's odd is that other reference based searches on Consent objects, such as organization, seem to behave as expected. For example, a search for Consents by organization behaves as expected

http://localhost:5555/fhir/Consent?organization=Organization/f001

**Environment:**
 - **OS**: macOS Catalina 10.15.7
 - **Browser**: Firefox Version: 83.0
 - **Docker**: 19.03.12
 - **Spark docker docker image**: sparkfhir/spark:r4-latest bbf8fed33cf6
 - **Spark mongo docker image**: sparkfhir/mongo:r4-latest 66a72babb60d

**Comment by kennethmyhra** [comment]
This was supposed to be fixed by this: https://github.com/FirelyTeam/spark/pull/324/commits/ae672e1e17e79430dc9afaf8731c5e12af7bf4dd

But as far as I can see the patient search parameter is not being indexed

@andy-a-o Could you look into why the search parameter `patient` is not indexed for the `Consent` resource

**Comment by kennethmyhra** [comment]
Fixed in version 1.5.3
